code/modules/projectiles/guns/flamer.dm

```diff
@@ -14,368 +14,302 @@
+			burn_list(turfs_skip_old)
+	if(!burn_list(turfs_to_ignite)) //These are so that both are called in the event one is false.
+		return
+	addtimer(CALLBACK(src, .proc/burn_stream, turfs_to_ignite, start_location, current_target, burn_type, range, FALSE), flame_spread_time)
+
+///Checks and lights the turfs in turfs_to_burn
+/obj/item/weapon/gun/flamer/proc/burn_list(list/turf/turfs_to_burn)
+	for(var/turf/turf_to_check in turfs_to_burn)
+		if((turf_to_check.density && !istype(turf_to_check, /turf/closed/wall/resin)) || isspaceturf(turf_to_check))
+			turfs_to_burn -= turf_to_check
 			continue
-		if((T.density && !istype(T, /turf/closed/wall/resin)) || isspaceturf(T))
-			break
-		if(!CHECK_BITFIELD(flags_item, IS_DEPLOYED) && loc != user && !master_gun)
-			break
-		if(!current_mag?.current_rounds)
-			break
-		var/range = istype(src, /obj/item/weapon/gun/flamer/mini_flamer) ? 4 : loaded_ammo.max_range //Temporary hardcode range of miniflamer.
-		if(distance > range)
-			break
+		for(var/obj/object in turf_to_check)
+			if(!object.density || object.throwpass || istype(object, /obj/structure/mineral_door/resin))
+				continue
+			turfs_to_burn -= turf_to_check
 
-		var/blocked = FALSE
-		for(var/obj/O in T)
-			if(O.density && !O.throwpass && !(O.flags_atom & ON_BORDER) && !istype(O, /obj/structure/mineral_door/resin))
-				blocked = TRUE
-				break
-
-		var/turf/TF
-		if(!prev_T.Adjacent(T) && (T.x != prev_T.x || T.y != prev_T.y)) //diagonally blocked, it will seek for a cardinal turf by the former target.
-			blocked = TRUE
-			var/turf/Ty = locate(prev_T.x, T.y, prev_T.z)
-			var/turf/Tx = locate(T.x, prev_T.y, prev_T.z)
-			for(var/turf/TB in shuffle(list(Ty, Tx)))
-				if(prev_T.Adjacent(TB) && ((!TB.density && !isspaceturf(T)) || istype(T, /turf/closed/wall/resin)))
-					TF = TB
-					break
-			if(!TF)
-				break
-		else
-			TF = T
+	if(!length(turfs_to_burn))
+		return FALSE
 
-		current_mag.current_rounds--
-		if (current_mag.current_rounds<=0)
-			light_pilot(user,FALSE)
-			break
-		flame_turf(TF,user, burntime, burnlevel, fire_color)
-		if(blocked)
-			break
-		distance++
-		prev_T = T
-		sleep(1)
-	if(!user)
-		return
-	user.hud_used.update_ammo_hud(user, src)
+	var/datum/ammo/flamethrower/loaded_ammo = CHECK_BITFIELD(flags_flamer_features, FLAMER_USES_GUN_FLAMES) ? ammo : current_mag.default_ammo
+	var/burn_level = initial(loaded_ammo.burnlevel) * burn_level_mod
+	var/burn_time = initial(loaded_ammo.burntime) * burn_time_mod
+	var/fire_color = initial(loaded_ammo.fire_color)
 
-/obj/item/weapon/gun/flamer/proc/flame_turf(turf/T, mob/living/user, heat, burn, f_color = "red")
-	if(!istype(T))
-		return
+	for(var/turf/turf_to_ignite in turfs_to_burn)
+		if(current_mag?.current_rounds <= 0)
+			light_pilot(FALSE)
+			return FALSE
+		flame_turf(turf_to_ignite, gun_user, burn_time, burn_level, fire_color)
+		current_mag.current_rounds--
+		gun_user?.hud_used.update_ammo_hud(gun_user, src)	
```

Comment:
no need to update every time

Reply:
i want the hud numbers to drain, itll look cool